### PR TITLE
support multiple layer yaml in fragments

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.SchemaDriven/OverwriteApplier.cs [hunk 1]
```diff
@@ -88,23 +88,71 @@ public object BuildOverwriteWithSchema(FileModel owModel, OverwriteDocumentModel
...
+            var transformed =  new Dictionary<string, object>
+            {
+                { OverwriteDocuments.Constants.FragmentsYAMLBlockKey, yamlBlockTransformed },
+                { OverwriteDocuments.Constants.FragmentsContentsKey, contentsTransformed },
+            };
+
+            AddSourceDetail(transformed, overwrite);
+            ReportDependencies(owModel, context);
+
+            return transformed;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/OverwriteApplier.cs]: extract some methods for duplicated code?

File: src/Microsoft.DocAsCode.Build.SchemaDriven/OverwriteApplier.cs [hunk 2]
```diff
@@ -88,23 +88,75 @@ public object BuildOverwriteWithSchema(FileModel owModel, OverwriteDocumentModel
...
+
+        public void MergeContentWithOverwrite(ref object source, object overwrite, string uid, string path, BaseSchema schema)
+        {
+            _merger.Merge(ref source, overwrite, uid, path, schema);
+        }
+
+        private object BuildClassicOverwriteWithSchema(FileModel owModel, OverwriteDocumentModel overwrite, BaseSchema schema)
+        {
+            ProcessContext context;
+            var transformed = BuildWithSchemaCore(overwrite.Metadata, owModel, overwrite, schema, out context) as IDictionary<string, object>;
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/OverwriteApplier.cs]: > as IDictionary<string, object> [](start = 110, length = 31)

no need

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/OverwriteApplier.cs]: >out context [](start = 98, length = 11)

seperate to another function to create context?

**Comment by 928PJY** [comment]
move to #2590

**Pull request closed without merging**